src/classes/PMT_Payment_TEST.cls

```diff
@@ -837,6 +837,871 @@ private class PMT_Payment_TEST {
+                AccountID = acc.Id,
+                CloseDate = System.today(),
+                npe01__Do_Not_Automatically_Create_Payment__c = true,
+                StageName = OPP_StageMappingUtil.TEST_FINALIZED_STAGE);
+        insert opp;
+
+        List<npe01__OppPayment__c> payments = new List<npe01__OppPayment__c> {
+                new npe01__OppPayment__c(
+                        npe01__Paid__c = true,
+                        npe01__Payment_Amount__c = 10,
+                        npe01__Opportunity__c = opp.Id,
+                        npe01__Payment_Date__c = System.today()
+                ),
+                new npe01__OppPayment__c(
+                        npe01__Paid__c = true,
+                        npe01__Payment_Amount__c = 10,
+                        npe01__Opportunity__c = opp.Id,
+                        npe01__Payment_Date__c = System.today()
+                )
+        };
+        insert payments;
+
+        Test.stopTest();
+
+        // check the opp amount
+        opp = [select Id, Amount from Opportunity where Id = :opp.Id];
+        System.assertEquals(20, opp.Amount, 'Finalized opportunity amount not updated when paid amounts are less than opportunity amount.');
+    }
+
+    /* Test with Enforce Accounting Data Consistency Enabled */
+    /* @description Non Finalized Opportunity, Amount changed to more than Sum of Paid Payments: Result (Allowed)
+    */
+    @isTest
+    private static void oppNonFinalizedUpdateAmtToHigherThanPaidPayments() {
+        UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_And_Orgs_Settings__c (npe01__Payments_Enabled__c = true, Enforce_Accounting_Data_Consistency__c = true));
+        Account acc = [SELECT Id FROM Account LIMIT 1];
+
+        Test.startTest();
+
+        Opportunity opp = new Opportunity(
+                Name='committed opp',
+                Amount = 100,
+                AccountID = acc.Id,
+                CloseDate = System.today(),
+                npe01__Do_Not_Automatically_Create_Payment__c = true,
+                StageName = OPP_StageMappingUtil.TEST_COMMITTED_STAGE);
+        insert opp;
+
+        List<npe01__OppPayment__c> payments = new List<npe01__OppPayment__c> {
+                new npe01__OppPayment__c(
+                        npe01__Paid__c = true,
+                        npe01__Payment_Amount__c = 50,
+                        npe01__Opportunity__c = opp.Id,
+                        npe01__Payment_Date__c = System.today()
+                ),
+                new npe01__OppPayment__c(
+                        npe01__Paid__c = true,
+                        npe01__Payment_Amount__c = 50,
+                        npe01__Opportunity__c = opp.Id,
+                        npe01__Payment_Date__c = System.today()
+                )
+        };
+        insert payments;
+        
+        Test.stopTest();
+
+        // update the opp to a larger amount
+        try {
+            opp.Amount = 300;
+            update opp;
```

Comment:
This is fine, but the other way to handle this (that won't throw a DML exception on error) is to do the following:

Database.SaveResult sr = Database.update(opp, false);

You can then interrogate the SaveResult to see if it was successful, and or look further at the errors returned.

What you did is fine, just mentioning it in case it resonates for another use case, or in the future.